homework/index1.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+{
+  function createAndAppend(name, parent, options = {}) {
+    const elem = document.createElement(name);
+    parent.appendChild(elem);
+    Object.keys(options).forEach(key => {
+      const value = options[key];
+      if (key === 'text') {
+        elem.textContent = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+
+  function main() {
+    const root = document.getElementById('root');
+    createAndAppend('select', root, { id: 'dropdown-select' });
+    createAndAppend('div', root, { id: 'details' });
+    createAndAppend('div', root, { id: 'contributors' });
+    fetchAsyncMode(HYF_REPOS_URL);
+  }
+
+  async function fetchAsyncMode(url) {
+    try {
+      const response = await fetch(url);
+      const responseJSON = await response.json();
+      responseJSON.sort(sortByName).forEach(repoDataObj => {
+        createAndAppend('option', document.getElementById('dropdown-select'), {
+          text: repoDataObj.name,
+        });
+      });
+      listenerSelect(responseJSON);
+    } catch (error) {
+      console.log(error);
+    }
+  }
+
+  // sort repositories
+  function sortByName(repoObjA, repoObjB) {
+    const nameA = repoObjA.name.toUpperCase(); // ignore upper and lowercase
+    const nameB = repoObjB.name.toUpperCase(); // ignore upper and lowercase
+    if (nameA < nameB) {
+      return -1;
+    }
+    if (nameA > nameB) {
+      return 1;
+    }
+    return 0;
+  }
+  // add an event when selecting a repository
+  function listenerSelect(listOfRepos) {
+    document.getElementById('dropdown-select').addEventListener('change', event => {
+      const selectedRepo = event.target.value;
+      const selectedData = listOfRepos.filter(repoData => repoData.name === selectedRepo)[0];
```

Comment:
This seems to be returning `undefined` every time. Think this is the source of the bug.

Reply:
The objects have a `.repository` field:

```
    document.getElementById('dropdown-select').addEventListener('change', event => {
      const selectedRepoName = event.target.value;
      const selectedRepoData = this.repos.filter(
        repoData => repoData.repository.name === selectedRepoName,
      )[0];
      const index = this.repos.indexOf(selectedRepoData);
      this.fetchContributorsAndRender(index);
    });
```